go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -512,15 +522,20 @@ func (c *Controller) RequestFullSync(nodeInfos []nodeInfo) {
 	// Resync node infos and node IP templates.
 	c.syncNodeInfos(nodeInfos)
 
-	// Resync services.
-	services, err := c.serviceLister.List(labels.Everything())
-	if err != nil {
-		klog.Errorf("Cached lister failed!? %v", err)
-		return
-	}
+	// Resync all services unless we're processing the initial node tracker sync (in which case
```

Comment:
so.. here the ordering is:
initialize: svccontroller.NewController->NewNodeTRacker->RequestFullSync->we check if initialNTSync is done which won't be done since that part of code is not touched yet right?

Then we have the:
Run code => after the initialize is done and we have the DNC, we do DNC.Run-> startSvcController->Run-> do the initial node tracker sync

I am unsure how and when we can hit RequestFullSync.NTInitialSync=true condition? It seems like the initial sync is always going to happen afterwards right? Perhaps I am missing something. I will huddle with you.

Reply:
I've moved the line where I set the flag to true further below and renamed it more simply `startupDone`, since it actually refers to whether the initial sync for the three handlers we have (node tracker, service, endpointslice) has been completed. I didn't move `c.startupDone=true` _after_ we start the workers because I'm a bit paranoid about the fact that we might miss node tracker events in the teeny tiny time frame of the for loop itself:
```
	klog.Info("Starting workers")
	for i := 0; i < workers; i++ {
		go wait.Until(c.worker, c.workerLoopPeriod, stopCh)
	}
```

Reply:
`I didn't move c.startupDone=true after we start the workers because I'm a bit paranoid about the fact that we might miss node tracker events in the teeny tiny time frame of the for loop itself`

ack that makes sense, but then there is also the reverse scene where for some nodes we get the add/update whatever between this tiny window where the done = true and workers are NOT Ready yet due to the loop not having finished but I think that should be negligible.